src/JWT/JwtBuilder.cs

```diff
@@ -0,0 +1,266 @@
+        /// </summary>
+        /// <returns>The JWT String.</returns>
+        public string Build()
+        {
+            if (!CanBuild())
+            {
+                throw new Exception("Can't build a Token. Check if you have call all of this: \n\r" +
+                                    "- SetAlgorithm \n\r" +
+                                    "- SetSerializer \n\r" +
+                                    "- SetUrlEncoder \n\r" +
+                                    "- SetSecret \n\r");
+            }
+            var encoder = new JwtEncoder(_algorithm, _serializer, _urlEncoder);
+            return encoder.Encode(_jwt.Payload, _secret);
+        }
+
+        /// <summary>
+        /// Decode a token with the decode-information you pushed in the buider.
+        /// </summary>
+        /// <param name="token">the JWT you want to extract</param>
+        /// <returns>return the json payload</returns>
+        public string Decode(string token)
+        {
+            TryToCreateAValidTor();
+            if (!CanDecode())
+            {
+                throw new Exception("Can't build a Token. Check if you have call all of this: \n\r" +
+                                    "- SetSerializer \n\r" +
+                                    "- SetUrlEncoder \n\r" +
+                                    "- SetTimeProvider \n\r" +
+                                    "- SetValidTor \n\r" +
+                                    "If you called MustVerify you must also call SetSecret"
+                );
+
+            }
+            var decoder = new JwtDecoder(_serializer, _validTor, _urlEncoder);
+            if (_verify == false)
+            {
+                return decoder.Decode(token);
+            }
+            return decoder.Decode(token, _secret, _verify);
+        }
+
+        /// <summary>
+        /// Decode a token with the decode-information you pushed in the buider.
+        /// </summary>
+        /// <param name="token">the JWT you want to Extract</param>
+        /// <returns>The payload converted to <see cref="T" /></returns>
+        public T Decode<T>(string token)
+        {
+            TryToCreateAValidTor();
+            if (!CanDecode())
+            {
+                throw new Exception("Can't build a Token. Check if you have call all of this: \n\r" +
+                                    "- SetSerializer \n\r" +
+                                    "- SetUrlEncoder \n\r" +
+                                    "- SetTimeProvider \n\r" +
+                                    "- SetValidTor \n\r" +
+                                    "If you called MustVerify you must also call SetSecret"
+                );
+
+            }
+            var decoder = new JwtDecoder(_serializer, _validTor, _urlEncoder);
+            return decoder.DecodeToObject<T>(token, _secret, _verify);
+        }
+
+        /// <summary>
+        /// Try to create a validTor is not a custom validTor set.
```

Comment:
Might want to do a find-replace for `validTor` 😄

Reply:
Yeap, done